backend/api/api.go

```diff
@@ -79,18 +83,47 @@ func Setup(redmineConf cfg.RedmineConfig) *fiber.App {
 		return c.SendStatus(200)
 	})
 
+	app.Get("/api/time_entries", func(c *fiber.Ctx) error {
+		apiKey, err := getSessionApiKey(c, store)
+		if err != nil {
+			log.Errorf("Failed to get session api key: %v", err)
+			return c.SendStatus(401)
+		}
+		timeEntries, err := redmine.GetTimeEntries(redmineConf, apiKey, c.Query("start", defaultDate), c.Query("end", defaultDate))
+		if err != nil {
+			log.Errorf("Failed to get recent time entries: %v", err)
+			c.Response().SetBodyString(err.Error())
+			return c.SendStatus(500)
+		}
+		return c.JSON(timeEntries)
+	})
+
 	app.Get("/api/issues", func(c *fiber.Ctx) error {
 		apiKey, err := getSessionApiKey(c, store)
 		if err != nil {
 			log.Errorf("Failed to get session api key: %v", err)
 			return c.SendStatus(401)
 		}
-		issuesJson, err := redmine.ListIssues(redmineConf, apiKey)
+		timeEntries, err := redmine.GetTimeEntries(redmineConf, apiKey, c.Query("start", defaultDate), c.Query("end", defaultDate))
+		if err != nil {
+			log.Errorf("Failed to get recent entries: %v", err)
+			c.Response().SetBodyString(err.Error())
+			return c.SendStatus(500)
+		}
+		var issueIds []string
+		for _, entry := range timeEntries.TimeEntries {
+			issueId := strconv.Itoa(entry.Issue.Id)
+			if !utl.Contains(issueIds, issueId) {
+				issueIds = append(issueIds, issueId)
+			}
+		}
+		issues, err := redmine.GetIssues(redmineConf, apiKey, issueIds)
```

Comment:
Issue IDs are ints, so converting the issue ID to a string seems unnecessary.

I wonder if there is a set storage type we could use for doing the "uniqifying" of the list of issue IDs so that we could just add them to a set and then pass them to `GetIssues` by converting the set to a list of IDs?

Reply:
I still feel it's odd to convert integers to strings. The backend database code expects and stores integers when it stores IDs.